File: app/src/main/java/br/fib/tentaasorte/Sorteio.java
package br.fib.tentaasorte;

import java.util.ArrayList;

public class Sorteio {

        private static Sorteio instance;
        private ArrayList<Cliente> clientes;

        private Sorteio() {
            this.clientes = new ArrayList<Cliente>();

            this.clientes.add( new Cliente("Will", "São Paulo"));
            this.clientes.add( new Cliente("Rosana", "Palmeiras"));
            this.clientes.add( new Cliente("Zé", "Santos"));
        }

        public static Sorteio getInstance() {
            if (instance == null) {
                instance = new Sorteio();
            }
            return instance;
        }

        public ArrayList<Cliente> ListarClientes() {
            return clientes;
        }

        public int NumerosVendidos(){
            return clientes.size();
        }

        public Cliente getCliente(int id ) {
            return clientes.get(id);
        }

        public void registrarVenda( String nome, String time ){
            this.clientes.add( new Cliente( nome, time) );
        }
    }

File: app/src/main/java/br/fib/tentaasorte/Sorteio.kt
package br.fib.tentaasorte

object Sorteio {

    private var listadetimes = arrayOf("Flamengo","Santos","Palmeiras","Grêmio","Athletico-PR","São Paulo","Internacional","Corinthians",
        "Fortaleza","Goiás","Bahia","Vasco","Atlético-MG","Fluminense","Botafogo","Ceará","Cruzeiro",
        "CSA","Chapecoense","Avaí")

    private val clientes: ArrayList<Cliente>

    init {
        this.clientes = ArrayList<Cliente>()
    }

    fun getQtdeDeTime(): Int {
        return listadetimes.size
    }

    fun verificarTimesDisponiveis(): ArrayList<String> {

        var times = ArrayList<String>()

        for (time in listadetimes){
            if ( !timeJaSelecionado( time ) )
            {
                times.add(time)
            }
        }

        return times
    }

    fun ListarClientes(): ArrayList<Cliente> {
        return this.clientes
    }

    fun NumerosVendidos(): Int {
        return this.clientes.size
    }

    fun getCliente(id: Int): Cliente? {
        return this.clientes.get(id)
    }

    fun registrarVenda(nome: String, time: String) {
        this.clientes.add( Cliente( nome, time ))
    }

    fun timeJaSelecionado( time: String ): Boolean {

        var arr = timesEscolhidos()

        return arr.contains(time)
    }

    fun timesEscolhidos(): ArrayList<String> {

        val arr = ArrayList<String>()

        for (i in this.clientes.indices){
            arr.add( this.clientes.get(i).getTime() )
        }

        return arr
    }

    fun limparListaDeTimeVendidos(){
        this.clientes.clear()
    }

    fun getNomeTime( id : Int ) : String {
        return listadetimes.get(id)
    }
}